src/components/FollowButton.test.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { fireEvent, render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import { FollowButton } from './FollowButton';
+
+describe('FollowButton', () => {
+  it('renders with default props', () => {
+    const tree = renderer
+      .create(<FollowButton followed onClick={console.log} />)
+      .toJSON();
+    expect(tree).toMatchInlineSnapshot(`
+      <div
+        className="raf-follow-button raf-follow-button--active"
+        onClick={[Function]}
+        role="button"
+      >
+        Following
+      </div>
+    `);
+  });
+
+  it('checks if onClick has been called', () => {
+    const onClick = jest.fn();
+
+    render(<FollowButton onClick={onClick} />);
+
+    fireEvent.click(screen.getByRole('button'));
+
+    expect(onClick).toHaveBeenCalledTimes(1);
+  });
+
+  it('renders same after "followed" prop change', () => {
+    const { rerender } = render(<FollowButton followed={false} />);
+
+    expect(screen.getByRole('button')).toHaveTextContent('Follow');
+
+    rerender(<FollowButton followed />);
+
+    expect(screen.getByRole('button')).toHaveTextContent('Follow');
```

Comment:
```suggestion
    expect(screen.getByRole('button')).toHaveTextContent('Following');
```

Reply:
I was weirded out by this:
![image](https://user-images.githubusercontent.com/43254280/109437982-04a2f100-7a28-11eb-9939-625c11017e36.png)

I thought that somebody designed this so that the value of the `followed` property is just initial but by closer inspection I realised that the state isn't used at all. What's weird about this though is that the test passes with current code. 🤔

Reply:
toHaveTextContent is probably doing a partial matching